src/test/java/com/google/jenkins/plugins/storage/IT/DownloadStepPipelineIT.java

```diff
@@ -0,0 +1,134 @@
+import com.google.api.client.http.InputStreamContent;
+import com.google.api.services.storage.Storage;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotPrivateKeyCredentials;
+import com.google.jenkins.plugins.credentials.oauth.ServiceAccountConfig;
+import com.google.jenkins.plugins.storage.StringJsonServiceAccountConfig;
+import hudson.EnvVars;
+import hudson.model.Result;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import java.io.InputStream;
+import java.net.URLConnection;
+import java.util.logging.Logger;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class DownloadStepPipelineIT {
+  private static final Logger LOGGER = Logger.getLogger(DownloadStepPipelineIT.class.getName());
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+  private static EnvVars envVars;
+  private static String projectId;
+  private static String credentialsId;
+  private static String bucket;
+  private static String pattern = "downloadstep_test.txt";
+
+  @BeforeClass
+  public static void init() throws Exception {
+    LOGGER.info("Initializing DownloadStepPipelineIT");
+
+    projectId = System.getenv("GOOGLE_PROJECT_ID");
+    assertNotNull("GOOGLE_PROJECT_ID env var must be set", projectId);
+    bucket = System.getenv("GOOGLE_BUCKET");
+    assertNotNull("GOOGLE_BUCKET env var must be set", bucket);
+
+    String serviceAccountKeyJson = System.getenv("GOOGLE_CREDENTIALS");
+    assertNotNull("GOOGLE_CREDENTIALS env var must be set", serviceAccountKeyJson);
+    credentialsId = projectId;
+    ServiceAccountConfig sac = new StringJsonServiceAccountConfig(serviceAccountKeyJson);
+    Credentials c = (Credentials) new GoogleRobotPrivateKeyCredentials(credentialsId, sac, null);
+    CredentialsStore store =
+        new SystemCredentialsProvider.ProviderImpl().getStore(jenkinsRule.jenkins);
+    store.addCredentials(Domain.global(), c);
+
+    EnvironmentVariablesNodeProperty prop = new EnvironmentVariablesNodeProperty();
+    envVars = prop.getEnvVars();
+    envVars.put("CREDENTIALS_ID", credentialsId);
+    envVars.put("BUCKET", bucket);
+    envVars.put("PATTERN", pattern);
+    jenkinsRule.jenkins.getGlobalNodeProperties().add(prop);
+
+    // create file to download
+    Storage service = getService(jenkinsRule.jenkins.get(), credentialsId);
+
+    InputStream stream = DownloadStepPipelineIT.class.getResourceAsStream(pattern);
+    String contentType = URLConnection.guessContentTypeFromStream(stream);
+    InputStreamContent content = new InputStreamContent(contentType, stream);
+    Storage.Objects.Insert insert = service.objects().insert(bucket, null, content);
```

Comment:
setName() actually returns the Insert that called it so these 3 lines can be converted to
```java
service.objects().insert(bucket, null, content).setName(pattern).execute();
```